test/test_room_api_view.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+import pytest
+from django.test import RequestFactory
+from rest_framework.response import Response
+
+from application.entities.room_entity import RoomEntity
+from application.repositories.interfaces import GetAllRooms
+from room.views import RoomAPIView
+
+
+@pytest.mark.django_db
+class TestRoomAPI:
+    def test_get_all_rooms_returns_empty_array_when_no_rooms_found(self, rf: RequestFactory):
```

Comment:
This test actually involves an actual DB. I would usually NOT write it like this, since a View should not care about the DB.